### ISSUE TITLE
Prysm ValidatorsByPubKey returns 0 balances for all but last pubkey

### ISSUE THREAD

**Issue by BisonDee** [attestantio/go-eth2-client#4]
Given the following 
```
if provider, isProvider := b.service.(ValidatorsProvider); isProvider {
    validators, err := provider.ValidatorsByPubKey(ctx, "head", pubkeys)
}
```

Only the first validator in the returned slice has the correct balance set, all the preceeding balances are set to 0

On Pyrmont I tested with the following keys
```
b1aa1fbe5851d7477ba12042f05bf406771471a118252bb1d455a184af23a4f317d854668f683aba629dcd3f698ba7b7
a7e6da76277e0ab2fbfc69ce532fa71cf7ea977102042ac5ed9b4ea6adf1346a56bce68d8f731d682d6eeaa8cab06cc8
90c39607ff913c77b1d5565143d2d75a16e07ef32c943ea055aef73af66c2d430c0f59980041bd99b55bac67dd597cf4
```
and got 
```
{
  "110293": {
    "index": "110293",
    "balance": "161027066386",
    "status": "Active_ongoing",
    "validator": {
      "pubkey": "0x90c39607ff913c77b1d5565143d2d75a16e07ef32c943ea055aef73af66c2d430c0f59980041bd99b55bac67dd597cf4",
      "withdrawal_credentials": "0x00ea4c7b221960571961b8e4b9dff539d650aaadbe85362f37879619dd5b872e",
      "effective_balance": "32000000000",
      "slashed": false,
      "activation_eligibility_epoch": "5094",
      "activation_epoch": "5174",
      "exit_epoch": "18446744073709551615",
      "withdrawable_epoch": "18446744073709551615"
    }
  },
  "110532": {
    "index": "110532",
    "balance": "0",
    "status": "Active_ongoing",
    "validator": {
      "pubkey": "0xa7e6da76277e0ab2fbfc69ce532fa71cf7ea977102042ac5ed9b4ea6adf1346a56bce68d8f731d682d6eeaa8cab06cc8",
      "withdrawal_credentials": "0x00575239c1b43b616da1bb476f3b0f730f24e867ec445d35a4aff4ae1898e358",
      "effective_balance": "32000000000",
      "slashed": false,
      "activation_eligibility_epoch": "5555",
      "activation_epoch": "5565",
      "exit_epoch": "18446744073709551615",
      "withdrawable_epoch": "18446744073709551615"
    }
  },
  "117132": {
    "index": "117132",
    "balance": "0",
    "status": "Active_ongoing",
    "validator": {
      "pubkey": "0xb1aa1fbe5851d7477ba12042f05bf406771471a118252bb1d455a184af23a4f317d854668f683aba629dcd3f698ba7b7",
      "withdrawal_credentials": "0x0077eff8b10549d5f8f5a0bd1a7a3e3bb679086489d5838bf27694647f85c3bd",
      "effective_balance": "32000000000",
      "slashed": false,
      "activation_eligibility_epoch": "16167",
      "activation_epoch": "16178",
      "exit_epoch": "18446744073709551615",
      "withdrawable_epoch": "18446744073709551615"
    }
  }
}
```

When passing one key at a time, I get the following results
```
{
  "117132": {
    "index": "117132",
    "balance": "192724520139",
    "status": "Active_ongoing",
    "validator": {
      "pubkey": "0xb1aa1fbe5851d7477ba12042f05bf406771471a118252bb1d455a184af23a4f317d854668f683aba629dcd3f698ba7b7",
      "withdrawal_credentials": "0x0077eff8b10549d5f8f5a0bd1a7a3e3bb679086489d5838bf27694647f85c3bd",
      "effective_balance": "32000000000",
      "slashed": false,
      "activation_eligibility_epoch": "16167",
      "activation_epoch": "16178",
      "exit_epoch": "18446744073709551615",
      "withdrawable_epoch": "18446744073709551615"
    }
  }
}
{
  "110532": {
    "index": "110532",
    "balance": "127042406111",
    "status": "Active_ongoing",
    "validator": {
      "pubkey": "0xa7e6da76277e0ab2fbfc69ce532fa71cf7ea977102042ac5ed9b4ea6adf1346a56bce68d8f731d682d6eeaa8cab06cc8",
      "withdrawal_credentials": "0x00575239c1b43b616da1bb476f3b0f730f24e867ec445d35a4aff4ae1898e358",
      "effective_balance": "32000000000",
      "slashed": false,
      "activation_eligibility_epoch": "5555",
      "activation_epoch": "5565",
      "exit_epoch": "18446744073709551615",
      "withdrawable_epoch": "18446744073709551615"
    }
  }
}
{
  "110293": {
    "index": "110293",
    "balance": "161027066386",
    "status": "Active_ongoing",
    "validator": {
      "pubkey": "0x90c39607ff913c77b1d5565143d2d75a16e07ef32c943ea055aef73af66c2d430c0f59980041bd99b55bac67dd597cf4",
      "withdrawal_credentials": "0x00ea4c7b221960571961b8e4b9dff539d650aaadbe85362f37879619dd5b872e",
      "effective_balance": "32000000000",
      "slashed": false,
      "activation_eligibility_epoch": "5094",
      "activation_epoch": "5174",
      "exit_epoch": "18446744073709551615",
      "withdrawable_epoch": "18446744073709551615"
    }
  }
}
```

**Comment by mcdee** [comment]
Thanks for raising this issue.  This is being addressed in #5  If you have a chance to test it please let me know how it goes, otherwise I'll roll it in to a new release.

**Comment by BisonDee** [comment]
Excellent, thanks :)
